test/local_integration_test.py

```diff
@@ -0,0 +1,182 @@
+        plugin = Plugin.load()
+        test_dss_query = copy.deepcopy(plugin.dss_subscription_query())
+        prefix_section = {
+            "prefix": {
+                "uuid": self.bundle_uuid_prefix
+            }
+        }
+        test_dss_query['query']['bool']['must'].append(prefix_section)
+        self.dss_query = test_dss_query
+
+    def tearDown(self):
+        subscribe(False)
+
+    def test_webservice_and_indexer(self):
+        subscribe(True)
+        query = self.dss_query
+        test_uuid = str(uuid.uuid4())
+        test_name = f'integration-test_{test_uuid}_{self.bundle_uuid_prefix}'
+        test_reindexer = Reindexer(indexer_url=config.indexer_endpoint(),
+                                   test_name=test_name,
+                                   es_query=query)
+
+        if False:
+            logger.info('Deleting all indices ...')
+            test_reindexer.delete_all_indices()
+
+        logger.info('Starting test using test name, %s ...', test_name)
+        logger.info('Creating indices and reindexing ...')
+        selected_bundle_fqids = test_reindexer.reindex()
+        self.num_of_bundles = len(selected_bundle_fqids)
+        self.check_bundles_are_indexed(test_name, 'files', set(selected_bundle_fqids))
+        self.check_endpoint_is_working('/')
+        self.check_endpoint_is_working('/health')
+        self.check_endpoint_is_working('/version')
+        self.check_endpoint_is_working('/repository/summary')
+        self.check_endpoint_is_working('/repository/files/order')
+        self.check_endpoint_is_working('/repository/files/export')
+
+    def check_endpoint_is_working(self, url: str):
+        url = config.service_endpoint() + url
+        response = requests.get(url)
+        response.raise_for_status()
+
+    def get_number_of_messages(self, queues: Dict[str, boto3.resources.base.ServiceResource]):
+        total_message_count = 0
+        for queue_name, queue in queues.items():
+            attributes = tuple('ApproximateNumberOfMessages' + suffix for suffix in ('', 'NotVisible', 'Delayed'))
+            num_messages = tuple(int(queue.attributes[attribute]) for attribute in attributes)
+            num_available, num_inflight, num_delayed = num_messages
+            queue.reload()
+            message_sum = sum((num_available, num_inflight, num_delayed))
+            logger.info('Queue %s has %i messages, %i available, %i in flight and %i delayed',
+                        queue_name, message_sum, num_available, num_inflight, num_delayed)
+            total_message_count += message_sum
+        logger.info('Total # of messages: %i', total_message_count)
+        return total_message_count
+
+    @property
+    def default_queue_empty_timeout(self):
+        return max(self.num_of_bundles * 30, 60)
+
+    def wait_for_queues_to_empty(self, queue_names: List[str], timeout: int = None):
```

Comment:
This method contains the same code twice. You should know by now that code duplication is generally a bad idea. This can be refactored into two calls of the same method `wait_for_queue_level(empty=True, timeout=…); wait_for_queue_level(empty=False, timeout=…)`.

Reply:
There is no point to the `queue_names` parameter. There is only one callsite and value passed to `queue_names` at that site is fixed.